src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/messages/IntentMsg.java

```diff
@@ -55,4 +56,17 @@ public IntentMsg(String requesterNode, String destinationNode, Map<String, Strin
   public String toString() {
     return String.format("Intent::from: '%s', to: '%s'", requesterNode, destinationNode);
   }
+
+  @Override
+  public boolean equals(final Object o) {
```

Comment:
If this is only for tests, can we can check for the `requestNode` and `destinationNode` strings explicitly instead of overriding the `equals` methods ? 
Otherwise in Java, with `equals()`, `hashcode()` needs to be implemented as well

Reply:
Nice catch! I'll address this issue in the next PR.

Reply:
This is a closed pull request ? I see that you closed it but then reopened it as well. Please consider closing it if it does not need to be reviewed any more.

Reply:
^ I closed it because the FileGCTest failed, so to get the CI check to pass, I had to close and reopen the PR because CI is triggered on a submission.

Reply:
I've removed the equals() methods on classes where they were unnecessary, and included hashcode() for classes where they were necessary.